### PR TITLE
docs: update formatting fields docs

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #8684 

### Description:

Created 2 sections

Added screenshots, info, and notes on formatting for Table calculations & Custom metrics

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: docs/docs/guides/formatting-your-fields.mdx [hunk 1]
```diff
@@ -20,32 +26,32 @@ In your dbt YAML file, it'll look something like this:
 ```yaml
 version: 2
 models:
-- name: users
-  columns:
-    - name: first_name
-      meta:
-        dimension:
-          hidden: true
+    - name: users
```
Comment (TuringLovesDeathMetal) [inline docs/docs/guides/formatting-your-fields.mdx:29]: I think this is adding too many spaces in the .yml? Normally it's 2-spaced tabs, not 4-spaced?

**Review by TuringLovesDeathMetal — CHANGES_REQUESTED** [review]
Nice! the additions look great. I just think your linter has added 4-spaced tabs instead of two-spaced tabs in all of the .yml snippets (which I'm not sure is actually correct? I think it's just 2 spaces?)

**Comment by joaoviana** [comment]
> Nice! the additions look great. I just think your linter has added 4-spaced tabs instead of two-spaced tabs in all of the .yml snippets (which I'm not sure is actually correct? I think it's just 2 spaces?)

oops :/ 

Thanks for checking this @TuringLovesDeathMetal , will revert this formatting 🤦

**Review by TuringLovesDeathMetal — APPROVED** [review]
looks great!

**Pull request merged**